flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/utils/ValidatorUtils.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.kubernetes.operator.utils;
+
+import org.apache.flink.kubernetes.operator.validation.DefaultDeploymentValidator;
+import org.apache.flink.kubernetes.operator.validation.FlinkDeploymentValidator;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.LinkedList;
+import java.util.List;
+
+/** Validator utilities. */
+public final class ValidatorUtils {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ValidatorUtils.class);
+
+    public static FlinkDeploymentValidator discoverValidator(
+            ClassLoader classLoader, String validatorIdentifier) {
+        final List<FlinkDeploymentValidator> validators = discoverValidators(classLoader);
+        return validators.stream()
+                .filter(
+                        v ->
+                                v.validatorIdentifier()
+                                        .toLowerCase()
+                                        .equals(validatorIdentifier.toLowerCase()))
+                .findFirst()
+                .orElse(DefaultDeploymentValidator.INSTANCE);
+    }
+
+    private static List<FlinkDeploymentValidator> discoverValidators(ClassLoader classLoader) {
+        final List<FlinkDeploymentValidator> result = new LinkedList<>();
+        ServiceLoaderUtils.load(FlinkDeploymentValidator.class, classLoader)
```

Comment:
Actually I am wondering maybe it would be better to use the plugin mechanism here similar to how metrics reporters are loaded:

```
 private static Iterator<MetricReporterFactory> getAllReporterFactories(@Nullable PluginManager pluginManager) {
        Iterator<MetricReporterFactory> factoryIteratorSPI = ServiceLoader.load(MetricReporterFactory.class).iterator();
        Iterator<MetricReporterFactory> factoryIteratorPlugins = pluginManager != null ? pluginManager.load(MetricReporterFactory.class) : Collections.emptyIterator();
        return Iterators.concat(factoryIteratorPlugins, factoryIteratorSPI);
    }
```

@wangyang0918 @morhidi wdyt?

Reply:
I am in favor of reusing the Flink plugin mechanism to load validator. This could avoid potential class conflicts if it is a uber jar without shading.